pkg/controllers/utils/podopslifecycle/utils.go

```diff
@@ -85,6 +87,16 @@ func Begin(c client.Client, adapter LifecycleAdapter, obj client.Object, updateF
 	return false, nil
 }
 
+// BeginWithCleaningOld is used for an CRD Operator to begin a lifecycle with cleaning the old lifecycle
+func BeginWithCleaningOld(c client.Client, adapter LifecycleAdapter, obj client.Object, updateFunc ...UpdateFunc) (updated bool, err error) {
+	if podInUpdateLifecycle, err := podopslifecycleutil.IsLifecycleOnPod(adapter.GetID(), obj); err != nil {
+		return false, fmt.Errorf("fail to check %s PodOpsLifecycle on Pod %s/%s: %s", adapter.GetID(), obj.GetNamespace(), obj.GetName(), err)
+	} else if podInUpdateLifecycle {
+		return false, Undo(c, adapter, obj)
```

Comment:
Undo, then Begin, and then return?

Reply:
https://github.com/KusionStack/kuperator/pull/295/commits/947ca3c8b1c14af6f76c5052a132f59718c472a2